src/pages/auth/index.tsx

```diff
@@ -1,11 +1,22 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
+import axios from "axios";
 import Link from "next/link";
+import Router from "next/router";
 
 import AuthHeading from "@/components/AuthHeading";
 import BaseButton from "@/components/base/BaseButton";
 import InputField from "@/components/base/InputField";
+import { useStore } from "../store/store";
 
 export default function Auth() {
+  const getCSRFCookie = async () => {
```

Comment:
nice! now please put this function outside of the component since otherwise you'll be creating a new one on every render for no reason